common/src/stack/command/stack/commands/load/json/__init__.py

```diff
@@ -0,0 +1,163 @@
+	MustBeRoot = 0
+	notifications = True
+
+
+class Command(command):
+	"""
+	Load configuration data from the provided json document. If no arguments
+	are provided then all plugins will be run.
+
+	<param optional='0' type='string' name='file'>
+	The json document containing the configuration information.
+	</param>
+
+	<arg optional='1' type='string' name='software'>
+	Load pallet, cart, and box data. If the pallet exists on a remote
+	server that requires authentication, be sure to provide the username
+	and password in the corresponding json fields.
+	</arg>
+
+	<arg optional='1' type='string' name='host'>
+	Load name, rack, rank, interface, attr, firewall, box, appliance,
+	comment, metadata, environment, osaction, route, group, partition,
+	and controller data for each host.
+	</arg>
+
+	<arg optional='1' type='string' name='network'>
+	Load name, address, gateway, netmask, dsn, pxe, mtu, and zone
+	data for each network.
+	</arg>
+
+	<arg optional='1' type='string' name='global'>
+	Load attr, route, firewall, partition, and controller data for
+	the global scope.
+	</arg>
+
+	<arg optional='1' type='string' name='os'>
+	Load name, attr, route, firewall, partition, and controller
+	data for each os.
+	</arg>
+
+	<arg optional='1' type='string' name='appliance'>
+	Load name, attr, route, firewall, partition, and controller
+	data for each appliance.
+	</arg>
+
+	<arg optional='1' type='string' name='group'>
+	Load name data for each group.
+	</arg>
+
+	<arg optional='1' type='string' name='bootaction'>
+	Load name, kernel, ramdisk, type, arg, and os data for each
+	bootaction.
+	</arg>
+	"""
+
+	def checks(self, args):
+
+		# Make sure that the user is not attepting to add more than one frontend
+		if (not args or 'host' in args) and 'host' in self.data:
+			current_frontend = json.loads(self.command('list.host', [ 'a:frontend', 'output-format=json' ]))[0]['host']
+			for host in self.data['host']:
+				if host['appliance'] == 'frontend' and host['name'] != current_frontend:
+					raise CommandError(self, 'frontend name in the json document does not match the current frontend name')
+
+		# Make sure that all of the pallets at least have URLs
+		if (not args or 'software' in args) and 'software' in self.data and 'pallet' in self.data['software']:
+			for pallet in self.data['software']['pallet']:
+				if pallet['url'] == None:
+					raise CommandError(self, f'pallet {pallet["name"]} {pallet["version"]} has no url')
+
+
+	# 'command' and 'parameters' are the same as a usual self.command call
+	# command_description is the string that will be printed in the logs, for example: f'adding host {hostname}'
+	# warning_string is used to filter CommandErrors, checking to see if the error is simply that the item already exists in the database
+	# the two warning_strings that are being used curently are 'already' and 'exists'
+	# try_command returns a return code to indicate if the command was successful
```

Comment:
If return codes are only ever 0/1, just return True/False.  Where True => Success without warnings, False => non-fatal warning, and fatal warnings don't return, they CommandError.